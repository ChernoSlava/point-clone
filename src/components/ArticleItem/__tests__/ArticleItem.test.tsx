import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";

import { ArticleItem } from "..";

describe("ArticleItem tests", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test("renders with correct title, image, and description", () => {
    const article = {
      id: "1",
      thumbnail: "test-thumbnail.jpg",
      title: "Test Title",
      description: "Test Description",
      date: "2 дня назад",
    };

    const { container, unmount } = render(
      <MemoryRouter>
        <ArticleItem
          id={article.id}
          title={article.title}
          thumbnail={article.thumbnail}
          description={article.description}
          date={article.date}
        />
      </MemoryRouter>
    );

    expect(screen.getByText(article.title)).toBeTruthy();
    expect(screen.getByAltText("Cartoon")).toBeTruthy();
    expect(screen.getByAltText("icon")).toBeTruthy();
    expect(screen.getByText(article.description)).toBeTruthy();

    expect(container).toMatchSnapshot();
    unmount();
  });
});
