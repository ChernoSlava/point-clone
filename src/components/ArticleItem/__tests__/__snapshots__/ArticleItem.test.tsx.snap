// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleItem tests renders with correct title, image, and description 1`] = `
<div>
  <li
    class="sc-beySPh jOqDaV"
  >
    <div
      class="sc-guDLey dSXgup"
    >
      <img
        alt="Cartoon"
        class="sc-dmyCSP gctnZZ"
      />
    </div>
    <div
      class="sc-hLQSwg ezfliH"
    >
      <h3
        class="sc-eDLKkx bCcaMp"
      >
        Test Title
      </h3>
      <p
        class="sc-jTQCzO gawhwJ"
      >
        Test Description
      </p>
      <p
        class="sc-iBdnpw giGobE"
      >
        <img
          alt="icon"
          class="sc-gLLuof gNHDaF"
          src="test-thumbnail.jpg"
        />
        2 дня назад
      </p>
    </div>
  </li>
</div>
`;
